src/swarm-engine/Swarm/Game/State/Robot.hs

```diff
@@ -374,7 +365,7 @@ wakeWatchingRobots myID currentTick loc = do
         [ (currentTick, currTickWakeable)
         , (addTicks 1 currentTick, nextTickWakeable)
         ]
-      newInsertions = M.filter (not . null) $ M.fromList wakeTimeGroups
+      newInsertions = MM.fromList wakeTimeGroups
```

Comment:
`MonoidMap.fromList` already arranges that the underlying support map has no mappings to `[]`, so there's no need to explicitly filter them out.